charmhelpers/contrib/charmsupport/nrpe.py

```diff
@@ -509,6 +510,37 @@ def add_haproxy_checks(nrpe, unit_name):
         check_cmd='check_haproxy_queue_depth.sh')
 
 
+def add_openvswitch_checks(nrpe, unit_name):
+    """
+    Add checks for openvswitch
+
+    :param NRPE nrpe: NRPE object to add check to
+    :param str unit_name: Unit name to use in check description
+    """
+    enable_sudo_for_openvswitch_checks()
+    nrpe.add_check(
+        shortname='openvswitch',
+        description='Check Open vSwitch {%s}' % unit_name,
+        check_cmd='check_openvswitch.py')
+
+
+def enable_sudo_for_openvswitch_checks():
+    sudoers_dir = "/etc/sudoers.d"
+    sudoers_mode = 0o100440
+    ovs_sudoers_file = "99-check_openvswitch"
+    ovs_sudoers_entry = "nagios ALL=(root) NOPASSWD: /usr/bin/ovs-vsctl show"
+    dest = os.path.join(sudoers_dir, ovs_sudoers_file)
+    try:
+        with open(dest, "w") as sudoer_file:
+            sudoer_file.write(ovs_sudoers_entry)
+        os.chmod(dest, sudoers_mode)
+        os.chown(dest, uid=0, gid=0)
+    except (OSError, IOError) as e:
+        log("Failed to setup sudoers file for check_openvswitch: {}".format(e))
+    else:
+        log("Sudoers file for check_openvswitch installed: {}".format(dest), DEBUG)
+
```

Comment:
Would it be possible to add unit tests for these two functions?

Reply:
I'll see if I can adapt the current unit testing around this module to also test these functions, however, my plan was to add functional testing to the consuming charms since these are both external callout functions and the tests would simply be to ensure the functions are called with expected inputs, since they'd all be mocked.

Reply:
I've added unittests and they are passing (though I'm seeing failures on test_check_restart_timestamps (tests.contrib.openstack.test_deferred_events.DeferredCharmServiceEventsTestCase))

```
  File "/home/ubuntu/charm-helpers/tests/contrib/openstack/test_deferred_events.py", line 326, in test_check_restart_timestamps
    self.assertFalse(clear_deferred_restarts.called)
AssertionError: True is not false
```

This is also erroring in stable/21.04 branch, so I'm ignoring for this PR.

Reply:
Looks like that error is my test env, but OSCI appears okay.